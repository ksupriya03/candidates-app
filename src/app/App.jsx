import React from 'react';
import { Provider } from 'react-redux';
import Routes from './Routes';
import Store from './Store';
import { BrowserRouter as Router } from 'react-router-dom';
import { createBrowserHistory } from 'history';

const history = createBrowserHistory();

const App = () => {
  return (
    <Provider store={Store}>
      <Router history={history}>
        <div>
          <Routes />
        </div>
      </Router>
    </Provider>
  );
};

export default App;
