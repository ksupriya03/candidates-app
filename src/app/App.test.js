import { render, screen } from '@testing-library/react';

import App from './App';

describe('<App />', () => {
  test('Snapshot test', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  test('renders `List of candidates` Dashboard header', () => {
    render(<App />);
    const linkElement = screen.getByText(/List of candidates/i);
    expect(linkElement).toBeInTheDocument();
  });
});
