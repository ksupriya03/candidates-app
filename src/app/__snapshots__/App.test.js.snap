// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Snapshot test 1`] = `
<div>
  <main>
    <div
      class="dashboard"
    >
      <table
        class="table table-hover"
      >
        <caption
          class="caption"
        >
          List of candidates
        </caption>
        <thead
          class="table-header"
        >
          <tr
            class="table-header table-heading"
          >
            <th
              class="table-heading-space"
              scope="col"
            >
              #
            </th>
            <th
              class="table-heading-space"
              scope="col"
            >
              Candidate Name
            </th>
            <th
              class="table-heading-space"
              scope="col"
            >
              Email Id
            </th>
            <th
              class="table-heading-space"
              scope="col"
            >
              JobID
            </th>
          </tr>
        </thead>
        <tbody
          class="table-header"
        >
          <tr
            class="table-header-row"
          >
            <td
              class="table-heading-space"
            >
              1
            </td>
            <td
              class="table-heading-space"
            >
              <a
                data-testid="John"
                href="/details/dummy-id"
              >
                John
              </a>
            </td>
            <td
              class="table-heading-space"
            >
              John@gmail.com
            </td>
            <td
              class="table-heading-space"
            >
              flybuys2231
            </td>
          </tr>
          <tr
            class="table-header-row"
          >
            <td
              class="table-heading-space"
            >
              2
            </td>
            <td
              class="table-heading-space"
            >
              <a
                data-testid="mary"
                href="/details/dummy-id"
              >
                mary
              </a>
            </td>
            <td
              class="table-heading-space"
            >
              mary@mail.com
            </td>
            <td
              class="table-heading-space"
            >
              flybuys222
            </td>
          </tr>
          <tr
            class="table-header-row"
          >
            <td
              class="table-heading-space"
            >
              3
            </td>
            <td
              class="table-heading-space"
            >
              <a
                data-testid="Jane"
                href="/details/dummy-id"
              >
                Jane
              </a>
            </td>
            <td
              class="table-heading-space"
            >
              jane@mail.com
            </td>
            <td
              class="table-heading-space"
            >
              flybuys123
            </td>
          </tr>
          <tr
            class="table-header-row"
          >
            <td
              class="table-heading-space"
            >
              4
            </td>
            <td
              class="table-heading-space"
            >
              <a
                data-testid="Dave"
                href="/details/dummy-id"
              >
                Dave
              </a>
            </td>
            <td
              class="table-heading-space"
            >
              dave@mail.com
            </td>
            <td
              class="table-heading-space"
            >
              flybuys222
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </main>
</div>
`;
